### PR TITLE
Added support for working with BufferHandler

### PR DESCRIPTION
**Pull request by gvozdb**
Added `handleBatch` method and fixed `write` method for `TelegramBotHandler`.

This is necessary to work with `BufferHandler`, `OverflowHandler`, etc.

### CODE REVIEW

**Comment by gvozdb** [comment]
Why checking failed – http://joxi.ru/Grq0DGEck5BxpA ? I don't understand.

**Comment by axell-brendow** [comment]
> Why checking failed – http://joxi.ru/Grq0DGEck5BxpA ? I don't understand.

Checking failed for PHP:nightly

**Comment by gvozdb** [comment]
@axell-brendow What is this? It's important?

File: src/Monolog/Handler/TelegramBotHandler.php [hunk 1]
```diff
@@ -122,12 +122,31 @@ public function disableNotification(bool $disableNotification = null): self
...
+        }
+    }
+
     /**
      * @inheritDoc
      */
     protected function write(array $record): void
     {
-        $this->send($record['formatted']);
+        $this->send($record['formatted'], [$record['formatted']]);
```
Comment (Seldaek) [inline src/Monolog/Handler/TelegramBotHandler.php:149]: I don't see how this fixes anything? The send method receives only one argument.

File: src/Monolog/Handler/TelegramBotHandler.php [hunk 2]
```diff
@@ -122,12 +122,31 @@ public function disableNotification(bool $disableNotification = null): self
...
+        foreach ($records as $record) {
+            if ($record['level'] < $this->level) {
+                continue;
+            }
+            $messages[] = $this->processRecord($record);
+        }
+
+        if (!empty($messages)) {
+            $this->send((string) $this->getFormatter()->formatBatch($messages), $messages);
+        }
```
Comment (Seldaek) [inline src/Monolog/Handler/TelegramBotHandler.php:141]: This should look more like the parent handle method IMO:

```suggestion
        $messages = [];

        foreach ($records as $record) {
            if (!$this->isHandling($record)) {
                continue;
            }

            if ($this->processors) {
                $record = $this->processRecord($record);
            }

            $messages[] = $record;
        }

        if (!empty($messages)) {
            $this->send((string) $this->getFormatter()->formatBatch($messages));
        }
```

File: src/Monolog/Handler/TelegramBotHandler.php (hunk 1 above)
Comment (gvozdb) [reply src/Monolog/Handler/TelegramBotHandler.php:149]: @Seldaek Yeah. I missed it. I've added some corrections to the next commit.

Comment (Seldaek) [reply src/Monolog/Handler/TelegramBotHandler.php:149]: I still don't get the point, $records isn't used in the send method, so why is this passed at all?

Comment (Seldaek) [reply src/Monolog/Handler/TelegramBotHandler.php:154]: Also handleBatch calls send with a single argument, so it's broken now that you added the second required argument.

**Comment by Seldaek** [comment]
Ok fixed things up and merged, thanks

**Pull request merged**